imblearn/utils/_validation.py

```diff
@@ -27,6 +27,42 @@
 TARGET_KIND = ("binary", "multiclass", "multilabel-indicator")
 
 
+class _OutputReconstructor:
+    """A class for converting input types to numpy and back."""
+
+    def __init__(self, X, y):
+        self.x_props = self._gets_props(X)
```

Comment:
It means then that we could have a dict to store the properties/

Reply:
We could see that when we recover #463

Reply:
Ok let wait for that. Anyway this is a private tool. We will be able to change the api